extensions/librdkafka/PublishKafka.cpp

```diff
@@ -407,7 +407,7 @@ class ReadCallback : public InputStreamCallback {
       return 0;
     }
 
-    for (size_t segment_num = 0; read_size_ < flow_size_; ++segment_num) {
+    for (size_t segment_num = 0; gsl::narrow<uint32_t>(read_size_) < flow_size_; ++segment_num) {
```

Comment:
Can we not actually change the `read_size_` type?

Reply:
I was wondering the same, but I couldn't decide if the value can go below zero, but I suppose it shouldn't. I will change the type.

Reply:
Updated in [75e1aa](https://github.com/apache/nifi-minifi-cpp/pull/1020/commits/75e1aa33766b2b43b135ebfd2aac312e796d463a)